### PR TITLE
fix Chain store takes tipset keys (#3119)

### LINKED ISSUE #3119: Chain store takes tipset keys
### Description
As pointed out [here](https://github.com/filecoin-project/go-filecoin/pull/3111#discussion_r306561887) the tipindex accessor methods on the chain store API take in strings in places where they should take in tipset keys.

### Acceptance criteria
Move to tipset keys in chain store.

### Risks + pitfalls
Should be very straightforward.

### Where to begin
`chain/store.go`: `HasTipSetAndState`, `GetTipSetAndStatesByParentsAndHeight`, `HasTipSetAndStatesWithParentsAndHeight`

### PR DESCRIPTION
**Pull request by deaswang**
Fix #3119

Add Serialize func for TipSetKey. It can replace String() if need.

### CODE REVIEW

File: types/tipset_key.go [hunk 1]
```diff
@@ -128,6 +131,17 @@ func (s TipSetKey) String() string {
 	return out + " }"
 }
 
+// Serialize returns a serialize string of TipSetKey use gob and base64.
```
Comment (ZenGround0) [inline types/tipset_key.go]: Blocking: We already have a canonical encoding for `TipSetKey` into cbor defined at the top of this file.  I don't see the value in defining a gob serialization like this right now.  Perhaps I'm missing something so feel free to explain why you think this is important.

File: types/tipset_key.go [hunk 2]
```diff
@@ -128,6 +131,17 @@ func (s TipSetKey) String() string {
...
+func (s TipSetKey) Serialize() string {
+	b := bytes.Buffer{}
+	e := gob.NewEncoder(&b)
+	err := e.Encode(s)
+	if err != nil {
+		return ""
```
Comment (frrist) [inline types/tipset_key.go]: in the off chance that we leave this method on `TipSetKey`, we should **not** be swallowing this error.

File: types/tipset_key.go (hunk 1 above)
Comment (deaswang) [reply types/tipset_key.go]: OK, Remove it. and the method just for review to if it need.
https://filecoinproject.slack.com/archives/CEHHJNJS3/p1564377470109500

**Comment by anorth** [comment]
Thanks, let's land this and then consider serialization later.

@ZenGround0 I requested a simpler serialisation for use as a database/map key - there's little reason to pay the cost of pretty-printed braces and base32 encoding for that use. This is not for exchange with other nodes.

However, such a serialisation could be encoded external to the tipset key, e.g. in the store module that intends to use it.

**Pull request merged**